src/Parser/DestructionWarlock/Modules/Features/ImmolateUptime.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+import Module from 'Parser/Core/Module';
+import Enemies from 'Parser/Core/Modules/Enemies';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+class ImmolateUptime extends Module {
+  static dependencies = {
+    enemies: Enemies,
+  };
+
+  suggestions(when) {
+    const immolateUptime = this.enemies.getBuffUptime(SPELLS.IMMOLATE_DEBUFF.id) / this.owner.fightDuration;
+    when(immolateUptime).isLessThan(0.9)
+      .addSuggestion((suggest, actual, recommended) => {
+        return suggest('Your Immolate uptime can be improved. Try to pay more attention to it as it provides a significant amount of Soul Shard Fragments over the fight and is also a big portion of your total damage.')
```

Comment:
One change would be to add in the `<SpellLink >` to the suggestion text on Immolate and potentially Soul Shards. The common practice has been to normally link any spell within the suggestion text, so I would want to keep it consistent.